models/programActivityLog.js

```diff
@@ -0,0 +1,39 @@
+const mongoose = require('mongoose')
+const { Schema } = mongoose
+
+module.exports = {
+	name: 'programActivityLog',
+
+	schema: {
+		date: {
+			type: String, // YYYY-MM-DD (UTC)
+			required: true,
+			index: true,
+		},
+
+		activity: {
+			improvementProject: {
+				type: [
+					{
+						programId: {
+							type: Schema.Types.ObjectId,
+							required: true,
+						},
+						solutionIds: {
+							type: [Schema.Types.ObjectId],
+							default: [],
+						},
+					},
+				],
+				default: [],
+			},
+		},
+	},
```

Comment:
@Prajwal17Tunerlabs improvementProjectStatus: {},   
this key is required. keep it {} by default. data team will add this data directly